src/coreclr/jit/llvmlower.cpp

```diff
@@ -687,6 +669,29 @@ void Llvm::insertProlog()
     }
 }
 
+void Llvm::lowerLocalsAfterNodes()
+{
+    for (unsigned lclNum = 0; lclNum < _compiler->lvaCount; lclNum++)
+    {
+        LclVarDsc* varDsc = _compiler->lvaGetDesc(lclNum);
+        if (varDsc->lvPromoted)
+        {
+            for (unsigned index = 0; index < varDsc->lvFieldCnt; index++)
+            {
+                LclVarDsc* fieldVarDsc = _compiler->lvaGetDesc(varDsc->lvFieldLclStart + index);
+
+                fieldVarDsc->lvIsStructField = false;
+                fieldVarDsc->lvParentLcl = BAD_VAR_NUM;
+                fieldVarDsc->lvIsParam = false;
+            }
+
+            varDsc->lvPromoted = false;
+            varDsc->lvFieldLclStart = BAD_VAR_NUM;
+            varDsc->lvFieldCnt = 0;
+        }
+    }
+}
+
```

Comment:
We were not stripping all promotion annotations, which lead to problems with SSA assuming parents use their (dependently promoted) fields. Since our model is that we don't have promotion after lowering, this strips all of it.